### DISCUSSION TITLE
[OP Stack Deployment] <Title>

### DISCUSSION THREAD

**Discussion by peacefulprogress25** [ethereum-optimism/developers#260] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Unable to verify contract, keeps on saying Invalid API key

### Logs

{"status":"0","message":"NOTOK","result":"Invalid API Key"}

**Comment by imtipi** [comment]
in the discord you said you are using @nomiclabs/hardhat-etherscan,can you try https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify instead if you just want to verify contract?

**Comment by sbvegan** [comment]
I'm pretty sure the etherscan api keys are different for https://optimistic.etherscan.io/ and https://etherscan.io/ too. Did you use op specific etherscan api keys for this process?

**Comment by peacefulprogress25** [comment]
yes, using the ones for optimistic etherscan only.

It is giving same error when we are doing a direct call here also
https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify

Are there any issues with the chain or api rn ?

**Reply by sbvegan** [reply]
Can you try and generate a new API key and try it with that one?
